### ISSUE TITLE
deb package build fails

### ISSUE THREAD

**Issue by dtrifiro** [iterative/dvc-s3-repo#42]
https://github.com/iterative/dvc-s3-repo/runs/6190101807?check_suite_focus=true#step:6:2164

```* Starting container dvc-s3-repo-deb with cmd: ['bash', '-c', "pip install './dvc[all]' && pip install -r dvc/scripts/build-requirements.txt && python dvc/scripts/build.py deb"]
Processing ./dvc
Installing build dependencies: started
Installing build dependencies: finished with status 'done'
Getting requirements to build wheel: started
Getting requirements to build wheel: finished with status 'done'
Installing backend dependencies: started
Installing backend dependencies: finished with status 'done'
Preparing metadata (pyproject.toml): started
Preparing metadata (pyproject.toml): finished with status 'error'
error: subprocess-exited-with-error

× Preparing metadata (pyproject.toml) did not run successfully.
│ exit code: 1
╰─> [34 lines of output]
Traceback (most recent call last):
File "/root/.pyenv/versions/3.8.3/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 363, in <module>
main()
File "/root/.pyenv/versions/3.8.3/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 345, in main
json_out['return_val'] = hook(**hook_input['kwargs'])
File "/root/.pyenv/versions/3.8.3/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 164, in prepare_metadata_for_build_wheel
return hook(metadata_directory, config_settings)
File "/tmp/pip-build-env-qpht9uwk/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 188, in prepare_metadata_for_build_wheel
self.run_setup()
File "/tmp/pip-build-env-qpht9uwk/overlay/lib/python3.8/site-packages/setuptools/build_meta.py", line 174, in run_setup
exec(compile(code, __file__, 'exec'), locals())
File "setup.py", line 3, in <module>
setup()
File "/tmp/pip-build-env-qpht9uwk/overlay/lib/python3.8/site-packages/setuptools/__init__.py", line 87, in setup
return distutils.core.setup(**attrs)
File "/tmp/pip-build-env-qpht9uwk/overlay/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 109, in setup
_setup_distribution = dist = klass(attrs)
File "/tmp/pip-build-env-qpht9uwk/overlay/lib/python3.8/site-packages/setuptools/dist.py", line 462, in __init__
_Distribution.__init__(
File "/tmp/pip-build-env-qpht9uwk/overlay/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 293, in __init__
self.finalize_options()
File "/tmp/pip-build-env-qpht9uwk/overlay/lib/python3.8/site-packages/setuptools/dist.py", line 886, in finalize_options
ep(self)
File "/tmp/pip-build-env-qpht9uwk/overlay/lib/python3.8/site-packages/setuptools_scm/integration.py", line 104, in infer_version
_assign_version(dist, config)
File "/tmp/pip-build-env-qpht9uwk/overlay/lib/python3.8/site-packages/setuptools_scm/integration.py", line 51, in _assign_version
_version_missing(config)
File "/tmp/pip-build-env-qpht9uwk/overlay/lib/python3.8/site-packages/setuptools_scm/__init__.py", line 106, in _version_missing
raise LookupError(
LookupError: setuptools-scm was unable to detect version for /dvc-s3-repo/dvc.

Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.

For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
[end of output]

note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.```

**Comment by dtrifiro** [comment]
Still happening for the 2.10.2 build

https://github.com/iterative/dvc-s3-repo/runs/6224167632?check_suite_focus=true

**Comment by efiop** [comment]
Probably just need to fetch tags in https://github.com/iterative/dvc-s3-repo/blob/master/download.py

**Comment by dtrifiro** [comment]
Fixed by #45
